prometheus/client.go

```diff
@@ -66,7 +65,28 @@ func (in *Client) GetSourceServices(namespace string, servicename string) (map[s
 	return routes, nil
 }
 
+// GetServiceMetrics returns a map of metrics (indexed by description) related to the provided service identified by its namespace and service name.
+// Returned map includes istio metrics and envoy health.
+// It returns an error on any problem.
+func (in *Client) GetServiceMetrics(namespace string, servicename string, duration string) (map[string]Metric, error) {
+	metricChan, errChan := make(chan namedMetric), make(chan error)
+	api := v1.NewAPI(in.p8s)
+	nbCalls := getServiceMetricsAsync(api, namespace, servicename, duration, metricChan, errChan)
+
+	metrics := make(map[string]Metric)
+	var err error
+	for i := 0; i < nbCalls; i++ {
+		select {
+		case namedMetric := <-metricChan:
+			metrics[namedMetric.name] = namedMetric.metric
+		case e := <-errChan:
+			err = e
+		}
+	}
+	return metrics, err
```

Comment:
Does it make sense to return items and errors to the caller? Or just the error? Is the intention to keep processing the items even if an error happens (and ignore missing values) ?

Reply:
Well I think it's nicer if we can provide the information we got even if some queries failed (like because of an unstable network, for instance)